PCPTool.v11/exe/SDKSample.cpp

```diff
@@ -4246,6 +4263,151 @@ Delete a user key from the PCP storage.
+    _In_reads_(argc) WCHAR* argv[]
+)
+/*++
+This function will enumerate all keys that are held on the PCPKSP for this user
+or in the machine context.
+--*/
+{
+    HRESULT hr = S_OK;
+    NCRYPT_PROV_HANDLE hProv = NULL;
+    NCryptKeyName* pKeyName = NULL;
+    PVOID pEnumState = NULL;
+    DWORD dwFlags[2] = {
+                        NCRYPT_SILENT_FLAG | NCRYPT_MACHINE_KEY_FLAG,
+                        NCRYPT_SILENT_FLAG };
+    NCRYPT_KEY_HANDLE hKey = NULL;
+    basic_regex<wchar_t> regexKeyName;
+    int matchingKeys = 0;
+
+    // Mandatory parameter: Key Name
+    if (argc > 2)
+    {
+        regexKeyName = basic_regex<wchar_t>(argv[2]);
+    }
+    else
+    {
+        wprintf(L"%s %s [key name]\n",
+            argv[0],
+            argv[1]);
+        hr = E_INVALIDARG;
+        goto Cleanup;
+    }
+
+    if (FAILED(hr = HRESULT_FROM_WIN32(NCryptOpenStorageProvider(
+        &hProv,
+        MS_PLATFORM_CRYPTO_PROVIDER,
+        0))))
+    {
+        goto Cleanup;
+    }
+
+    for (UINT32 n = 0; n < (sizeof(dwFlags) / sizeof(DWORD)); n++)
+    {
+        hr = S_OK;
+
+        while (SUCCEEDED(hr))
+        {
+            hr = HRESULT_FROM_WIN32(NCryptEnumKeys(
+                hProv,
+                NULL,
+                &pKeyName,
+                &pEnumState,
+                dwFlags[n]));
+            if (FAILED(hr))
+            {
+                if (hr == HRESULT_FROM_WIN32((ULONG)NTE_NO_MORE_ITEMS))
+                {
+                    if (pEnumState != NULL)
+                    {
+                        NCryptFreeBuffer(pEnumState);
+                        pEnumState = NULL;
+                    }
+                    hr = S_OK;
+                    break;
+                }
+                else
+                {
+                    goto Cleanup;
+                }
+            }
+            else
+            {
+                if (regex_match(pKeyName->pszName, regexKeyName)) {
+                    matchingKeys++;
+
+                    if (FAILED(hr = HRESULT_FROM_WIN32(NCryptOpenKey(
+                        hProv,
+                        &hKey,
+                        pKeyName->pszName,
+                        0,
+                        dwFlags[n] & NCRYPT_MACHINE_KEY_FLAG))))
+                    {
+                        wprintf(L"Could not open key: %s (0x%08lx)\n", pKeyName->pszName, hr);
+                        if (pKeyName != NULL)
+                        {
+                            NCryptFreeBuffer(pKeyName);
+                            pKeyName = NULL;
+                        }
+                        hr = S_OK;
+                        continue;
+                    }
+
+                    wprintf(L"Deletes key: %s ", pKeyName->pszName);
+                    // Delete the key
+                    if (FAILED(hr = (NCryptDeleteKey(hKey, 0))))
+                    {
+                        wprintf(L" Failed (0x%08lx)\n", hr);
+                    }
+                    else {
```

Comment:
Could you uniform styles? Here ```{``` is on same line as ```else```. On others its on next line.

Reply:
Done!